File: 02-primes-sieve-of-eratosthenes/dotnet/PrimesSieve.cs
public static class PrimesSieve
{
    public static long[] FindPrimeNumbersSieve(long max)
    {
        var b = new bool[max];
        var primes = new List<long>();

        for (long i = 2; i < max; i++)
        {
            if (b[i])
            {
                continue;
            }

            primes.Add(i);

            for (var k = i * i; k < max; k += i)
            {
                b[k] = true;
            }
        }

        return primes.ToArray();
    }
}

File: 02-primes-sieve-of-eratosthenes/go/primessieve.go
package primessieve

func findPrimeNumbersSieveOfEratosthenes(max int) []int {
	b := make([]bool, max)

	var primes []int

	for i := 2; i < max; i++ {
		if b[i] {
			continue
		}

		primes = append(primes, i)

		for k := i * i; k < max; k += i {
			b[k] = true
		}
	}

	return primes
}
